variables.tf

```diff
@@ -0,0 +1,212 @@
+#####################################################
+# VPC Variables
+# Copyright 2020 IBM
+#####################################################
+
+variable "create_vpc" {
+  description = "True to create new VPC. False if VPC is already existing and subnets or address prefixies are to be added"
+  type        = bool
+}
+
+##############################################################################
+# Account Variables
+##############################################################################
+
+variable "region" {
+  description = "Region for resources to be created"
+  type        = string
+  default     = "us-south"
+}
+
+variable "unique_prefix" {
+  description = "Unique prefix for the network resources"
+  type        = string
+}
+
+variable "resource_group" {
+  description = "Name of resource group to create VPC. If not given, `Default` resource group will be used to create the network resources"
+  type        = string
+  default     = null
+}
+
+
+#####################################################
+# Optional Parameters
+#####################################################
+
+variable "vpc_name" {
+  description = "Name of the vpc"
+  type        = string
+  default     = null
+}
+
+variable "classic_access" {
+  description = "Classic Access to the VPC"
+  type        = bool
+  default     = null
+}
+
+variable "default_address_prefix" {
+  description = "Default address prefix creation method"
+  type        = string
+  default     = null
+}
+
+variable "default_network_acl_name" {
+  description = "Name of the Default ACL"
+  type        = string
+  default     = null
+}
+
+variable "default_security_group_name" {
+  description = "Name of the Default Security Group"
+  type        = string
+  default     = null
+}
+
+variable "default_routing_table_name" {
+  description = "Name of the Default Routing Table"
+  type        = string
+  default     = null
+}
+
+variable "vpc_tags" {
+  description = "List of Tags for the vpc"
+  type        = list(string)
+  default     = []
+}
+
+variable routing_table_list {
+  description = "List of Routing tables"
+  type        = list(string)
+  default     = []
+}
+
+variable routing_table_route_list {
+  description = "List of Routing table routes"
+  type        = list(string)
+  default     = []
+}
+
+variable "enable_gateway" {
+  description = "True to create new Gateway and enable it to subnets"
+  type        = bool
+  default     = false
+}
+
+variable "floating_ip" {
```

Comment:
Can we define the content types of map

Reply:
since floating ip can nhave either IP ID or address.. we can't define content type of map.